docs/shared/recipes/plugins/project-graph-plugins.md

```diff
@@ -21,113 +32,215 @@ You can register a plugin by adding it to the plugins array in `nx.json`:
 }
 ```
 
-## Implementing a Project Graph Processor
+## Adding New Nodes to the Project Graph
 
-A Project Graph Processor takes a project graph and returns a new project graph. It can add/remove nodes and edges.
+You can add nodes to the project graph with `createNodes`. This is the API that Nx uses under the hood to identify Nx projects coming from a `project.json` file or a `package.json` that's listed in a package manager's workspaces section.
 
-Plugins should export a function named `processProjectGraph` that handles updating the project graph with new nodes and edges. This function receives two things:
+The `createNodes` tuple is defined as follows:
```

Comment:
I would link to the actual reference documentation (/packages/devkit/documents/CreateNodesFunction) since that allows them to explore all the subtypes and will always be in sync with the actual code.